code/online_challenges/src/project_euler/problem_019.java

```diff
@@ -0,0 +1,91 @@
+import java.util.HashMap;
+import java.util.Map;
+
+class Problem019 {
+    public static void main(String args[]){
+        Map<Integer, String> L = new HashMap<Integer, String>();
+        L.put(1, "Sun");
+        L.put(2, "Mon");
+        L.put(3, "Tue");
+        L.put(4, "Wed");
+        L.put(5, "Thurs");
+        L.put(6, "Fri");
+        L.put(7, "Sat");
+     
+        // Start at Monday, 1st, 1900
+        int counter = 1;
+        int tally = 0;
+         
+        // Year
+        for (int yr = 1900; yr < 2001;  ++ yr)
+        {
+            System.out.println("Year = " + yr);
+            // Month
+            for (int month = 1; month < 13; month ++)
+            {
+                int y = findDays(month, yr);
+                // Number of month days
+                for (int monthDays = 1; monthDays <= y; monthDays ++) 
+                {
+                    // Start at Monday, 1st, 1900
+                    counter += 1;
+                    if ((monthDays == 1) && (L.get(counter) == "Sun"))
+                    {
+                        System.out.println("month_days = " + monthDays);
+                        tally += 1;
+                    }
+                    // Days of the week
+                    if (counter >= 7)
+                    {
+                        counter = 0;
+                    }
+                }
+            }
+            System.out.println();
+        }
+        // subtract 2 for the year 1900 - start counting at 1901
+        System.out.println("tally = " + (tally - 2));
```

Comment:
This `2` seems kind of hardcoded (where did it come from?). If you don't want to count the year 1900 you can just increment `tally` only if `yr != 1900` in line 35.